File: 268.missing-number.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#

# @lc code=start
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        max_num = len(nums)
        expected_total = (1 + max_num) * max_num // 2

        for val in nums:
            expected_total -= val
        return expected_total
# @lc code=end

File: 268.missing-number.java
/*
 * @lc app=leetcode id=268 lang=java
 *
 * [268] Missing Number
 */

// @lc code=start
class Solution {
    public int missingNumber(int[] nums) {
        int maxNum = nums.length;
        int leftSum = (1 + maxNum) * maxNum / 2;

        for (int val : nums) {
            leftSum -= val;
        }

        return leftSum;
    }
}
// @lc code=end

